metadata-jobs/mae-consumer/src/main/java/com/linkedin/metadata/kafka/MetadataChangeLogProcessor.java

```diff
@@ -63,53 +63,58 @@ public MetadataChangeLogProcessor(List<MetadataChangeLogHook> metadataChangeLogH
-        "Got Generic MCL on topic: {}, partition: {}, offset: {}",
-        consumerRecord.topic(),
-        consumerRecord.partition(),
-        consumerRecord.offset());
-    MetricUtils.counter(this.getClass(), "received_mcl_count").inc();
+    try (Timer.Context i = MetricUtils.timer(this.getClass(), "consume").time()) {
+      kafkaLagStats.update(System.currentTimeMillis() - consumerRecord.timestamp());
+      final GenericRecord record = consumerRecord.value();
+      log.debug(
+          "Got Generic MCL on topic: {}, partition: {}, offset: {}",
+          consumerRecord.topic(),
+          consumerRecord.partition(),
+          consumerRecord.offset());
+      MetricUtils.counter(this.getClass(), "received_mcl_count").inc();
+
+      MetadataChangeLog event;
+      try {
+        event = EventUtils.avroToPegasusMCL(record);
+        log.debug(
+            "Successfully converted Avro MCL to Pegasus MCL. urn: {}, key: {}",
+            event.getEntityUrn(),
+            event.getEntityKeyAspect());
+      } catch (Exception e) {
+        MetricUtils.counter(this.getClass(), "avro_to_pegasus_conversion_failure").inc();
+        log.error("Error deserializing message due to: ", e);
+        log.error("Message: {}", record.toString());
+        return;
+      }
 
-    MetadataChangeLog event;
-    try {
-      event = EventUtils.avroToPegasusMCL(record);
       log.debug(
-          "Successfully converted Avro MCL to Pegasus MCL. urn: {}, key: {}",
+          "Invoking MCL hooks for urn: {}, key: {}",
           event.getEntityUrn(),
           event.getEntityKeyAspect());
-    } catch (Exception e) {
-      MetricUtils.counter(this.getClass(), "avro_to_pegasus_conversion_failure").inc();
-      log.error("Error deserializing message due to: ", e);
-      log.error("Message: {}", record.toString());
-      return;
-    }
-
-    log.debug(
-        "Invoking MCL hooks for urn: {}, key: {}",
-        event.getEntityUrn(),
-        event.getEntityKeyAspect());
 
-    // Here - plug in additional "custom processor hooks"
-    for (MetadataChangeLogHook hook : this.hooks) {
-      if (!hook.isEnabled()) {
-        continue;
-      }
-      try (Timer.Context ignored =
-          MetricUtils.timer(this.getClass(), hook.getClass().getSimpleName() + "_latency").time()) {
-        hook.invoke(event);
-      } catch (Exception e) {
-        // Just skip this hook and continue. - Note that this represents "at most once" processing.
-        MetricUtils.counter(this.getClass(), hook.getClass().getSimpleName() + "_failure").inc();
-        log.error("Failed to execute MCL hook with name {}", hook.getClass().getCanonicalName(), e);
+      // Here - plug in additional "custom processor hooks"
+      for (MetadataChangeLogHook hook : this.hooks) {
+        if (!hook.isEnabled()) {
+          continue;
+        }
+        try (Timer.Context ignored =
+            MetricUtils.timer(this.getClass(), hook.getClass().getSimpleName() + "_latency")
```

Comment:
@anshbansal just a nit, i think there's a class to handle the name concatenation, eg. `MetricRegistry.name(this.getClass(), ...)`

Reply:
Also maybe no need to `_latency`? It will already have some suffix indicating it's a timer (e.g. `_75thPercentile`)